src/Core/Services/Implementations/UserService.cs

```diff
@@ -1244,6 +1245,18 @@ public async Task<bool> IsLegacyUser(string userId)
         return IsLegacyUser(user);
     }
 
+    public async Task<bool> IsManagedByAnyOrganizationAsync(Guid userId)
+    {
+        var organization = await _organizationRepository.GetByClaimedUserDomainAsync(userId);
+        if (organization is { Enabled: true })
+        {
+            var plan = StaticStore.GetPlan(organization.PlanType);
+            return plan.ProductTier == ProductTierType.Enterprise;
+        }
```

Comment:
This also needs updating to 
* use applicationCacheService
* refer to `useSso`
* flatten nesting

Reply:
I can't use `IApplicationCacheService` here, I need to use `OrganizationRepository.GetByClaimedUserDomainAsync` to get the organization that has a verified domain that matches the User's email domain

Reply:
You're right! I misread it.